### PR TITLE
Capstone project submission PR

### PR DESCRIPTION
**Pull request by bilalrajput09**
# Capstone project submission PR
✅Completed all tasks.
✅Each team member has contributed to their tasks.
✅Managed and work through the Kanban board.

### CODE REVIEW

File: src/modules/likeShow.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const likesShow = (likeID, category) => {
+  const [filteredLikeID] = likeID.filter((idObj) => category.idCategory === idObj.item_id);
```
Comment (rloterh) [inline src/modules/likeShow.js:2]: - [ ] Good job for adding counter logic 👌 However, let's ensure to indicate the counter value on how homepage, around the nav section, like seen in the [template design](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md)

**Example of item counter value on homepage nav section:**
image


**Your nav area:**
image

File: src/modules/commentsCounter.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    data.forEach((d) => {
+      commentHtml += `<p>${d.creation_date} ${d.username}: ${d.comment}</p>`;
+      counter += 1;
+    });
+    return counter;
+  }
+
+  return false;
+};
+
```
Comment (rloterh) [inline src/modules/commentsCounter.js:14]: - [ ] Similarly, let's ensure to indicate the comments counter value on the comment modal, like seen in the [template design](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md)

**Example of comment counter value on comment modal section:**
image

**Your comments modal:**
image

File: src/modules/reservationCounter.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      commentHtml += `<p>${d.date_start} - ${d.date_end} by ${d.username}</p>`;
+      counter += 1;
+    });
+
+    return counter;
+  }
+
+  return false;
+};
+
```
Comment (rloterh) [inline src/modules/reservationCounter.js:15]: - [ ] Again, let's ensure to indicate the reservation counter value on the reservation modal, like seen in the [template design](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md)

**Example of reservation counter value on reservation modal section:**
image

**Your comments reservation modal:**
image)

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    heartBtn.forEach((heart) => {
+      heart.addEventListener('click', (e) => {
+        const { id } = e.target.closest('.item_contianer');
+        heart.classList.add('colour_red');
+        let likeCounter = e.target.previousElementSibling.textContent;
+        likeCounter = +likeCounter;
+        likeCounter += 1;
+        e.target.previousElementSibling.textContent = String(likeCounter);
+        postLikes(id, projectID);
+      });
```
Comment (rloterh) [inline src/index.js]: - [x]  OPTIONAL
When I continue to click on the like button, it keeps the counter only keeps increasing. However, I think you can improve upon your _likes_ feature this way:
    - Like icon should toggle to indicate _like_ or _unlike_ when a particular user clicks on the like icon.
    - Total count of likes per user should be 1.

Can you try achieving this with the help of local storage?

File: src/style.css [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+.name_like_conatiner {
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 1em;
+  font-size: 20px;
+  font-weight: 700;
+  width: 70%;
+  align-self: center;
+}
```
Comment (rloterh) [inline src/style.css:90]: - [ ] OPTIONAL
To improve the **user experience** it will be good practice to ensure that the _Likes_ buttons, and similar other buttons are behaving as buttons when the mouse hovers on them. We can improve upon this aspect by adding the CSS attribute `cursor: pointer` to the _Comment_ button on _mouse hover_ events.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Team, 

You did extremely great on this project! 👏 👏
There are some issues that you still need to work on to prepare your project for the assessment but you are almost there 
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳 🏃‍♀️ 🏘️ 🏃‍♀️

## Highlights 👍
- Counter values are stored and fetched correctly using Involvement API!
- Nice job adding counters!
- Awesome UI design, matches wireframe design!

## Suggested changes
_Kindly check my inline comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._ 😃 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/modules/likeShow.js (hunk 1 above)
Comment (Bellagirl-maker) [reply src/modules/likeShow.js:2]: @rloterh done

File: src/index.js (hunk 4 above)
Comment (Bellagirl-maker) [reply src/index.js]: @rloterh done

File: src/style.css (hunk 5 above)
Comment (bilalrajput09) [reply src/style.css:90]: Done ✌️.

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+const commentsShow = (data) => {
+  commentHtml = '';
+  commentsCounter = 0;
+  if (!data) return;
+  data.forEach((d) => {
+    commentHtml += `<p>${d.creation_date} ${d.username}: ${d.comment}</p>`;
+    commentsCounter += 1;
+  });
+};
```
Comment (Zuhaib042) [inline src/index.js:36]: - [ ] Well done implementing the comments counter as suggested by the previous reviewer, but there seems to be one small issue. In the requirements, it says that the screen should get updated when the user clicks the button. now when the user clicks the button, the comment appears but the counter doesn't get updated. it gets updated when you click to open the popup the next time. the reason for that is you are calling this `commentsShow` function in the `createPopupWindow` function so commentsShow only gets called when createPopupWindow gets called. kindly consider fixing this small issue. I think that the comment should get updated right after the user comments.
 -  When the user clicks on the "Comment" button, the data is recorded in the Involvement API, and the screen is updated.
your comments popup
https://user-images.githubusercontent.com/96683843/234916517-017f95c9-0263-4519-b976-cd1d9f423234.mov

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+const reservationShow = (data) => {
+  commentHtml = '';
+  reservationCounter = 0;
+  if (!data) return;
+  data.forEach((d) => {
+    commentHtml += `<p>${d.date_start} - ${d.date_end} by ${d.username}</p>`;
+    reservationCounter += 1;
+  });
+};
```
Comment (Zuhaib042) [inline src/index.js:46]: - [ ] Well done implementing the reservation counter as suggested by the previous reviewer, but there seems to be the same small issue as it was in the comments counter. In the requirements, it says that the screen should get updated when the user clicks the button. now when the user clicks the button, the reservation appears but the counter doesn't get updated. it gets updated when you click to open the popup the next time. the reason for that is you are calling this reservationShow function in the createPopupWindowReservation function so reservationShow only gets called when createPopupWindowReservation gets called. kindly consider fixing this small issue. I think that the counter should get updated right after the user clicks on the reserve button.
- When the user clicks on the "Reserve" button, the data is recorded in the Involvement API and the screen is updated.

https://user-images.githubusercontent.com/96683843/234917986-b47b0f44-96f4-45f9-9601-0268d08db915.mov

File: tests/itemCounter.test.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  const dummyData = [{ item: '1' }, { item: '2' }];
+
+  expect(itemcounter(dummyData)).toBe(2);
+});
+
+test('Should return correct count for items', () => {
+  const dummyData = [{ item: '1' }];
+
+  expect(itemcounter(dummyData)).toBe(1);
+});
```
Comment (Zuhaib042) [inline tests/itemCounter.test.js]: - [OPTIONAL] Kindly consider thinking about a scenario where there is 0 item on the page so that the itemCounter function should return 0. Kindly consider adding a test for this edge case.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi team @bilalrajput09 @Bellagirl-maker @AbenezerAssefa 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✅ Implemented most of the suggested changes
✅ App is working perfectly

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 6 above)
Comment (bilalrajput09) [reply src/index.js:36]: Implemented ✅✅. Thanks for the suggestions✌️.

File: src/index.js (hunk 7 above)
Comment (bilalrajput09) [reply src/index.js:46]: Implemented ✅✅. Thanks for the suggestions✌️.

File: tests/itemCounter.test.js (hunk 8 above)
Comment (bilalrajput09) [reply tests/itemCounter.test.js]: Implemented ✅✅. Thanks for the suggestions✌️.

**Review by DammyShittu — APPROVED** [review]
Hi @bilalrajput09, @Bellagirl-maker and @AbenezerAssefa,

## APPROVED 🟢

## Highlights
✔️ Counters for comments and reservations

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
